### PR TITLE
add categories admin page

### PR DESCRIPTION
**Pull request by tri-et**

### CODE REVIEW

File: server/schema/category/def.js [hunk 1]
```diff
@@ -4,12 +4,15 @@ const Category = `
     name: String
     desc: String
     img: String
+    nameMainCategory: String
```
Comment (hemvanh) [inline server/schema/category/def.js]: nameMainCategory ???
--> must be:  mainCategoryName

File: server/schema/category/res.js [hunk 2]
```diff
@@ -1,11 +1,47 @@
...
     async listCategories(_, {input}) {
       let mainCategory = new MainCategory({id: input})
       return await mainCategory.getCategories()
     },
+    async fetchAllCategoriesAdmin() {
+      return await Category.findAll()
+    },
+  },
+  RootMutation: {
+    async createCategory(_, {input}) {
```
Comment (hemvanh) [inline server/schema/category/res.js]: need authAdmin ???

File: src/store/category/actions.js [hunk 3]
```diff
@@ -23,3 +24,104 @@ export const fetchRecs = ({commit}, payload) => {
       _procError(err)
     })
 }
+export const fetchCategorys = ({commit}) => {
```
Comment (hemvanh) [inline src/store/category/actions.js]: Categorys  -> Categories

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: src/store/category/actions.js (hunk 3 above)
Comment (tri-et) [reply src/store/category/actions.js]: can't...if change like that it'll have error in EtGrid
return dispatch(`${this.type}/fetch${upperFirst(this.type)}s`, payload)
if change to Categories must modify EtGrid again

Comment (hemvanh) [reply src/store/category/actions.js]: ETGrid already modify to have a aliasType name, so you could put  the aliasType="categorie"
http://prntscr.com/mlu5kw

**Pull request merged**